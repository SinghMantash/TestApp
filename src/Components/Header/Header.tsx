import {StyleSheet, Text, View} from 'react-native';
import React from 'react';
import colors from '../../styles/colors';
import {scaleHeight} from '../../styles/dimensions';

const Header = ({title=""}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.headerText} testID='header-text'>{title}</Text>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.secondaryColor,
    height: scaleHeight(50),
    alignItems: 'center',
    justifyContent: 'center',
  },
  headerText: {
    fontSize: 25,
    color: colors.backgroundColor,
  },
});
