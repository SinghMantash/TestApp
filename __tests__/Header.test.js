import 'react-native';
import React from 'react';

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Header from '../src/Components/Header/Header';
import {render, screen} from '@testing-library/react-native';
import {
  toBeEmptyElement,
  toHaveTextContent,
} from '@testing-library/jest-native';

expect.extend({toBeEmptyElement, toHaveTextContent});

it(' Header Snapshot Matching', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("Header's title Prop working ", async () => {
  const headerTitle = 'Testing';
  render(<Header title={headerTitle} />);
  const headerText = await screen.findByTestId('header-text');
  expect(headerText).toHaveTextContent(headerTitle);
});
