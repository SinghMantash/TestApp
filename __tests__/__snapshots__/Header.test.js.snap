// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Header Snapshot Matching 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F27900",
      "height": 82.14285714285714,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 25,
      }
    }
    testID="header-text"
  />
</View>
`;
